### PR TITLE
fix(extra): dockerfile errors and tsc fallback

### PR DESCRIPTION
**Pull request by SrIzan10**
hi

### CODE REVIEW

File: templates/extra/Dockerfile.TS.sern [hunk 1]
```diff
@@ -8,6 +8,6 @@ RUN npm install
 
 COPY . .
 
-RUN tsc --build
+RUN tsc --build || npx tsc --build
```
Comment (jacoobes) [inline templates/extra/Dockerfile.TS.sern:11]: add -p flag maybe to let tsc use local tsconfig?

```tsc -p "./tsconfig.json" --build || npx tsc -p "./tsconfig.json" --build

**Review by jacoobes — CHANGES_REQUESTED** [review]

File: templates/extra/Dockerfile.TS.sern (hunk 1 above)
Comment (SrIzan10) [reply templates/extra/Dockerfile.TS.sern:11]: doesnt it choose automatically?

Comment (SrIzan10) [reply templates/extra/Dockerfile.TS.sern:11]: I'm doing this because most of the times `tsc` isn't installed automatically in the dockerfile.

Comment (jacoobes) [reply templates/extra/Dockerfile.TS.sern:11]: from my experience not adding -p flag messes up build

**Comment by SrIzan10** [comment]
now that we're there we should just run `npm i -g typescript` for ppl like me who forget to add it as a devDependency or as a CLI command

**Comment by jacoobes** [comment]
do you think adding as a devDependency in docker is better and then we can just do

tsc || npx tsc "./tsconfig.json" --build

**Comment by EvolutionX-10** [comment]
Does this still need to be worked on or to be reviewed @jacoobes ?

**Comment by jacoobes** [comment]
yeah, still looks mergeable

**Comment by EvolutionX-10** [comment]
will rebase this soon after I complete my work

**Comment by EvolutionX-10** [comment]
@SrIzan10 could you re-create this in new pr please, looks like rebasing will be hell 💀

**Comment by SrIzan10** [comment]
it can still be mergeable, the dockerfiles hasn't been modified since last year

**Review by EvolutionX-10 — APPROVED** [review]

**Review by jacoobes — CHANGES_REQUESTED** [review]
Need to ensure we're using typescript and not deprecated tsc npm package 

https://stackoverflow.com/a/55138585

**Review by jacoobes — APPROVED** [review]

**Review by EvolutionX-10 — APPROVED** [review]

**Pull request merged**